src/activity_handlers.cpp

```diff
@@ -2728,6 +2730,50 @@ void activity_handlers::try_sleep_finish( player_activity *act, player *p )
     act->set_to_null();
 }
 
+void activity_handlers::build_do_turn( player_activity *act, player *p )
+{
+    item *con_item = act->targets.front().get_item();
+    // this shouldn't happen, unfinished crafts should not dissappear during activity
+    if( !con_item ) {
+        add_msg( m_bad, "The marker item is no longer there, cancelling construction," );
```

Comment:
I think this should be a debug message, this should actually never happen instead of hopefully never happen like with in progress crafts